ledger/complete/wal/checkpointer.go

```diff
@@ -185,11 +185,15 @@ func (c *Checkpointer) Checkpoint(to int, targetWriter func() (io.WriteCloser, e
 		return fmt.Errorf("cannot replay WAL: %w", err)
 	}
 
+	c.wal.log.Info().Msgf("flattening forest for checkpoint %d", to)
+
```

Comment:
@zhangchiqing thanks for making great suggestions!

I agree it would be useful to log progress during an operation that takes 6+ minutes.  However, in this case, logging progress requires some discussion and would be simpler if we handle that in a separate issue/pr.  

For example, we don't have access to when each WAL segment is loaded because it is done inside prometheus, etc. which gets in the way of logging segment progress.

Mind if we merge this PR as-is and open a separate issue to add more progress logging so we can iron out more details?

Reply:
yup, sounds good. Thanks for the details